### PR TITLE
Import 1password SDK, and make ExternalVariables.Get async

### PR DESCRIPTION
**Pull request by rain-on**
This was massive.

When we get an externalvariable, it MAY need to go "off-box" to fetch the variable, thus it should be asynchronous - this has necessitated a LARGE amount of change across the code base.

[sc-76784]

### CODE REVIEW

**Review by EddyMoulton — APPROVED** [review]
Couple of things for you to check/confirm - but LGTM

File: source/Calamari.Testing/EnvironmentVariables.cs [hunk 1]
```diff
@@ -22,25 +26,25 @@ public enum ExternalVariable
...
 
-        [EnvironmentVariable("K8S_OctopusAPITester_Token", "GKS Kubernetes API Test Cluster Token")]
+        [EnvironmentVariable("K8S_OctopusAPITester_Token", "GKS Kubernetes API Test Cluster Token", "op://Calamari Server Secrets for Tests/GKS Kubernetes API Test Cluster/Token")]
         KubernetesClusterToken,
 
-        [EnvironmentVariable("K8S_OctopusAPITester_Server", "GKS Kubernetes API Test Cluster Url")]
+        [EnvironmentVariable("K8S_OctopusAPITester_Server", "GKS Kubernetes API Test Cluster Url", "op://Calamari Server Secrets for Tests/GKS Kubernetes API Test Cluster/Server")]
         KubernetesClusterUrl,
 
         [EnvironmentVariable("Helm_OctopusAPITester_Password", "Artifactory Test Account")]
```
[the remark this answers is not in the document]
Comment (rain-on) [reply source/Calamari.Testing/EnvironmentVariables.cs:41]: For now - we're going to stay using Env Vars.
Then add entries to the vault (and remove env-vars) - and confirm its all working.
Then add remaining to the vault, and we remove env-vars from TC.
Then get all the vault entries rotated to be value specific to our team (and not shared etc.)

File: source/Calamari.Terraform.Tests/CommandsFixture.cs [hunk 2]
```diff
@@ -848,7 +874,11 @@ async Task<TestCalamariCommandResult> ExecuteAndReturnResult(string command, Act
                                                                   context.Variables.Add(TerraformSpecialVariables.Action.Terraform.CustomTerraformExecutable,
                                                                                         customTerraformExecutable);
 
-                                                                  populateVariables(context);
+                                                                  var task = populateVariables(context);
+                                                                  if (!task.IsCompleted)
+                                                                  {
+                                                                      task.RunSynchronously();
+                                                                  }
```
[the remark this answers is not in the document]
Comment (rain-on) [reply source/Calamari.Terraform.Tests/CommandsFixture.cs:881]: the problem is - sometimes, the function is already complete - eg if it doesn't contain an awai - and if you cann RunSync on an already completed task, it throws - this is a way to manage the async/sync without modifying too much down the chain.

File: source/Calamari.Testing/EnvironmentVariables.cs [hunk 3]
```diff
@@ -70,31 +74,45 @@ public enum ExternalVariable
         [EnvironmentVariable("CALAMARI_AUTHPASSWORD", "OctopusMyGetTester")]
         MyGetFeedPassword,
 
-        [EnvironmentVariable("GOOGLECLOUD_OCTOPUSAPITESTER_JSONKEY", "GoogleCloud - OctopusAPITester")]
+        [EnvironmentVariable("GOOGLECLOUD_OCTOPUSAPITESTER_JSONKEY", "GoogleCloud - OctopusAPITester", "op://Calamari Secrets for Tests/Google Cloud Octopus Api Tester JsonKey")]
```
[the remark this answers is not in the document]
Comment (rain-on) [reply source/Calamari.Testing/EnvironmentVariables.cs:77]: will resolve in the next PR :)

**Pull request merged**